src/deploy/functions/runtimes/discovery/index.ts

```diff
@@ -0,0 +1,98 @@
+import fetch from "node-fetch";
+import * as fs from "fs";
+import * as path from "path";
+import * as yaml from "js-yaml";
+import { promisify } from "util";
+
+import { logger } from "../../../../logger";
+import * as api from "../../.../../../../api";
+import * as backend from "../../backend";
+import * as runtimes from "..";
+import * as v1alpha1 from "./v1alpha1";
+import { FirebaseError } from "../../../../error";
+
+export const readFileAsync = promisify(fs.readFile);
+
+export function yamlToBackend(
+  yaml: any,
+  project: string,
+  region: string,
+  runtime: runtimes.Runtime
+): backend.Backend {
+  try {
+    if (!yaml.specVersion) {
+      throw new FirebaseError("Expect backend yaml to specify a version number");
+    }
+    if (yaml.specVersion === "v1alpha1") {
```

Comment:
If we want to follow AIP [stability levels](https://google.aip.dev/181) this should be `v1alpha` not `v1alpha1`...do you intend for the spec version to match API version, or do you think they should be versioned separately? I can see argument for both.